variables.tf

```diff
@@ -175,6 +175,18 @@ variable "groups" {
       enabled = bool
       rule    = string
     }))
+    role_assignments = optional(map(object({
+      name                                   = optional(string)
+      scope                                  = optional(string) #Sets scope to current subscription if empty
+      role_definition_id                     = optional(string)
+      role_definition_name                   = optional(string) #Either this or the above must be provided
+      assign_to_workload_principal           = optional(bool)
```

Comment:
Is there a reason to include `assign_to_workload_principal` on a group role assignment?